src/main/java/at/hannibal2/skyhanni/features/fishing/FishingTimer.kt

```diff
@@ -3,126 +3,220 @@ package at.hannibal2.skyhanni.features.fishing
+        } else if (config.fishingCapAlert && currentCount >= currentCap.getValue()) {
+            playSound()
         }
+    }
 
-        if (!rightLocation) return
+    private fun playSound() = SoundUtils.repeatSound(250, 4, SoundUtils.plingSound)
 
-        if (event.isMod(5)) checkMobs()
-        if (event.isMod(7)) tryPlaySound()
-        if (config.manualResetTimer.isKeyHeld() && Minecraft.getMinecraft().currentScreen == null) {
-            startTime = SimpleTimeMark.now()
+    @SubscribeEvent
+    fun onMobSpawn(event: MobEvent.Spawn.SkyblockMob) {
+        if (!isEnabled()) return
+        val mob = event.mob
+        if (babyMagmaSlugsToFind != 0 && mob.name == "Baby Magma Slug") {
+            recentBabyMagmaSlugs += mob
+            handleBabySlugs()
+            return
         }
+        if (mob.name !in SeaCreatureManager.allFishingMobs) return
+        recentMobs += mob
+        handle()
     }
 
-    private fun tryPlaySound() {
-        if (currentCount == 0) return
-
-        val passedSince = startTime.passedSince()
-        val barnTimerAlertTime = (config.alertTime * 1_000).milliseconds
-        if (passedSince in barnTimerAlertTime..(barnTimerAlertTime + 3.seconds)) {
-            SoundUtils.playBeepSound()
+    @SubscribeEvent
+    fun onMobDeSpawn(event: MobEvent.DeSpawn.SkyblockMob) {
+        if (!isEnabled()) return
+        val mob = event.mob
+        if (mob in mobMap) {
+            mobMap -= mob
+            if (mob.name == "Magma Slug") {
+                lastMagmaSlugLocation = mob.baseEntity.getLorenzVec()
+                babyMagmaSlugsToFind += 3
+                lastMagmaSlugTime = SimpleTimeMark.now()
+                handleBabySlugs()
+            }
         }
+        recentMobs -= mob
+        updateInfo()
     }
 
-    private fun checkMobs() {
-        val newCount = countMobs()
+    @SubscribeEvent
+    fun onSeaCreatureFish(event: SeaCreatureFishEvent) {
+        if (!isEnabled()) return
+        if (!rightLocation) return
+        lastSeaCreatureFished = SimpleTimeMark.now()
+        lastNameFished = event.seaCreature.name
+        mobsToFind = if (event.doubleHook) 2 else 1
+        handle()
+    }
 
-        if (currentCount == 0 && newCount > 0) {
-            startTime = SimpleTimeMark.now()
+    private fun handle() {
+        if (lastSeaCreatureFished.passedSince() > 2.seconds) return
+        val name = lastNameFished ?: return
+        val mobs = recentMobs.filter { it.name == name && it !in mobMap }
+            .sortedBy { it.baseEntity.distanceToPlayer() }
+            .take(mobsToFind).ifEmpty { return }
+        mobsToFind -= mobs.size
+        mobs.forEach { mobMap[it] = SimpleTimeMark.now() }
+        if (mobsToFind == 0) {
+            recentMobs.clear()
+            lastNameFished = null
         }
+        updateInfo()
+    }
 
-        currentCount = newCount
-        if (newCount == 0) {
-            startTime = SimpleTimeMark.farPast()
+    private fun handleBabySlugs() {
```

Comment:
why do we need to handle baby slugs differently? can this get moved to mob detection?

Reply:
discussed on discord